DataConnectors/O365 DataCSharp/Teams.CustomConnector.Common/KeyVaultHelper.cs

```diff
@@ -59,15 +59,15 @@ private static async Task<string> LoadKeysFromVault(string key)
             try
             {
                 var baseUrl = Environment.GetEnvironmentVariable(Constants.KeyVaultBaseUrl);
-                KeyVaultClient keyVaultClient = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(new AzureServiceTokenProvider().KeyVaultTokenCallback));
-                SecretBundle secretBundle = null;
+                SecretClient secretClient = new SecretClient(new Uri(baseUrl), new DefaultAzureCredential());
+                KeyVaultSecret keyVaultSecret = null;
 
                 await retryWithExponentialBackoff.RunAsync(async () =>
                 {
-                    secretBundle = await keyVaultClient.GetSecretAsync($"{baseUrl}/secrets/{key}");
+                    keyVaultSecret = await secretClient.GetSecretAsync(key);
                 });
 
-                return secretBundle?.Value;
+                return keyVaultSecret?.Value;
```

Comment:
Nit: the null check is unnecessary - an exception would've been thrown - but it doesn't hurt either.

Reply:
@heaths - I have updated, please review.